learn/work/sparse-matrix/index.qmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Model tuning using a sparse matrix"
+categories:
+ - tuning
+ - classification
+ - sparse data
+type: learn-subsection
+weight: 1
+description: | 
+  Fitting a model using tidymodels with a sparse matrix as the data.
+toc: true
+toc-depth: 2
+include-after-body: ../../../resources.html
+---
+
+```{r}
+#| label: "setup"
+#| include: false
+#| message: false
+#| warning: false
+source(here::here("common.R"))
+```
+  
+```{r}
+#| label: "load"
+#| include: false
+library(tidymodels)
+library(sparsevctrs)
+
+pkgs <- c("tidymodels", "sparsevctrs")
+
+theme_set(theme_bw() + theme(legend.position = "top"))
+```
+
+## Introduction
+
+`r article_req_pkgs(pkgs)`
+
+This article demonstrates how we can use a sparse matrix in tidymodels.
+
+## Example data
+
+The data we will be using in this article is a larger sample of the [small_fine_foods](https://modeldata.tidymodels.org/reference/small_fine_foods.html) data set from the [modeldata](https://modeldata.tidymodels.org) package. Data was downloaded from <https://snap.stanford.edu/data/web-FineFoods.html>, sliced down to 100,000 rows, tokenized, and saved as a sparse matrix. Data has been saved as [reviews.rds](reviews.rds) and the code to generate this data set is found at [generate-data.R](generate-data.R). This file takes up around 1MB compressed, and around 12MB once loaded into R.
+
+This data set is encoded as a sparse matrix from the Matrix package. We are using this data for this article because if we were to turn it into a dense matrix it would take up 3GB which is a considerable size.
```

Comment:
```suggestion
The data we will be using in this article is a larger sample of the [small_fine_foods](https://modeldata.tidymodels.org/reference/small_fine_foods.html) data set from the [modeldata](https://modeldata.tidymodels.org) package. The [raw data](https://snap.stanford.edu/data/web-FineFoods.html) was sliced down to 100,000 rows, tokenized, and saved as a sparse matrix. Data has been saved as [reviews.rds](reviews.rds) and the code to generate this data set is found at [generate-data.R](generate-data.R). This file takes up around 1MB compressed, and around 12MB once loaded into R. This data set is encoded as a sparse matrix from the Matrix package; if we were to turn it into a dense matrix, it would take up 3GB.
```